numpy/core/src/umath/simd.inc.src

```diff
@@ -85,19 +116,20 @@ abs_ptrdiff(char *a, char *b)
  * We instead rely on i32gather/scatter_ps instructions which use a 32-bit index
  * element. The index needs to be < INT_MAX to avoid overflow. MAX_STEP_SIZE ensures this.
  */
-#define IS_BINARY_SMALL_STEPS \
+#define IS_BINARY_SMALL_STEPS_AND_NOMEMOVERLAP \
     ((abs(steps[0]) < MAX_STEP_SIZE)  && \
      (abs(steps[1]) < MAX_STEP_SIZE)  && \
-     (abs(steps[2]) < MAX_STEP_SIZE))
+     (abs(steps[2]) < MAX_STEP_SIZE)  && \
+     (nomemoverlap(args[0], steps[0]*dimensions[0], args[2], steps[2]*dimensions[0])) && \
+     (nomemoverlap(args[1], steps[1]*dimensions[0], args[2], steps[2]*dimensions[0])))
 
 /*
  * output should be contiguous, can handle strided input data
  * Input step should be smaller than MAX_STEP_SIZE for performance
  */
```

Comment:
Comment needs an update for non-overlap condition

Reply:
Done.